### PR TITLE
Add Base2ExponentialHistogram example

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4505

Also some opportunistic Go Doc Comment fixes.

### CODE REVIEW

**Comment by pellared** [comment]
CC @arukiidou

**Review by dashpole — APPROVED** [review]

File: sdk/metric/example_test.go [hunk 1]
```diff
@@ -220,3 +220,23 @@ func ExampleNewView_wildcard() {
...
+func ExampleNewView_exponentialHistogram() {
+	// Create a view that makes the "latency" instrument
+	// to be reported as an exponential histogram.
+	view := metric.NewView(
+		metric.Instrument{
+			Name:  "latency",
+			Scope: instrumentation.Scope{Name: "http"},
+		},
+		metric.Stream{
+			Aggregation: metric.AggregationBase2ExponentialHistogram{},
```
Comment (MrAlias) [inline sdk/metric/example_test.go:233]:
```suggestion
			Aggregation: metric.AggregationBase2ExponentialHistogram{
				MaxSize:   160,
				MaxScale: 20,
			},
```

**Comment by arukiidou** [comment]
Thank you!
Right example gives user's learning curve improves

File: sdk/metric/example_test.go (hunk 1 above)
Comment (pellared) [reply sdk/metric/example_test.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4518/commits/638577858593723947f918f38e8c65642886546e

**Comment by pellared** [comment]
I also added the suggestion from https://github.com/open-telemetry/opentelemetry.io/pull/3275#discussion_r1327273102

**Pull request merged**